src/ring_theory/derivation.lean

```diff
@@ -88,12 +88,19 @@ protected lemma map_zero : D 0 = 0 := map_zero D
 @[simp] lemma map_smul : D (r • a) = r • D a := D.to_linear_map.map_smul r a
 @[simp] lemma leibniz : D (a * b) = a • D b + b • D a := D.leibniz' _ _
 
+@[simp, priority 900] lemma map_smul_of_tower {S : Type*} [has_scalar S A] [has_scalar S M]
+  [linear_map.compatible_smul A M S R] (r : S) : D (r • a) = r • D a :=
+D.to_linear_map.map_smul_of_tower r a
```

Comment:
Could you please repeat `D` and `a` as parameters, because now they will appear before `S` and all those typeclasses in the signature, which feels a bit messy to me.